xbmc/addons/AddonManager.cpp

```diff
@@ -598,6 +598,10 @@ bool CAddonMgr::FindAddon(const std::string& addonId, const AddonVersion& addonV
   FindAddons(installedAddons, "special://xbmc/addons");
   FindAddons(installedAddons, "special://home/addons");
 
+  std::set<std::string> installed;
+  for (const auto& addon : installedAddons)
+    installed.insert(addon.second->ID());
```

Comment:
Sorry, this is definitely wrong. `FindAddons` reads whatever is in the file system not taking the "installed" state into account. What is in the file system is not necessarily what is currently installed. Install of multiple addons runs concurrently! The file system in this scenario may contain a mixture of addons already installed and addons about to be installed. This is exactly what caused the bug fixed by #18499.